### PR TITLE
add proc stock to repositories

### PR DESCRIPTION
**Pull request by veolys-nanne**

### CODE REVIEW

File: Repository/AbstractAggregateRepository.php [hunk 1]
```diff
@@ -25,6 +25,30 @@ public function setAggregationQueryBuilder($aggregationQueryBuilder)
     }
 
     /**
```
Comment (itkg-alavieille) [inline Repository/AbstractAggregateRepository.php]: add @param string $criteria

File: Repository/AbstractAggregateRepository.php [hunk 2]
```diff
@@ -25,6 +25,30 @@ public function setAggregationQueryBuilder($aggregationQueryBuilder)
...
+     * @return array
+     */
+    public function findWithTransverseCriteria($criteria)
+    {
+        return $this->findBy($this->createTransverseCriteria($criteria));
+    }
+
+    /**
+     * @return string
```
Comment (alavieille) [inline Repository/AbstractAggregateRepository.php]: add @param string $criteria

File: Repository/AbstractAggregateRepository.php [hunk 3]
```diff
@@ -25,6 +25,30 @@ public function setAggregationQueryBuilder($aggregationQueryBuilder)
...
+     */
+    public function createTransverseCriteria($criteria)
+    {
+        $documentManager = $this->getDocumentManager();
+        $documentName = $this->getDocumentName();
+        $metaData = $documentManager->getClassMetadata($documentName);
+        $collectionName = $metaData->getCollection();
+        $dataBase = $documentManager->getDocumentDatabase($documentName);
+
+        $return = $dataBase->execute('db.loadServerScripts();return selectEnumeration({collection: "' . $collectionName . '", criteria: ' . $criteria . ');');
```
Comment (alavieille) [inline Repository/AbstractAggregateRepository.php]: you forgot `}` at the end

**Comment by alavieille** [comment]
can you add a test ?

**Comment by veolys-nanne** [comment]
test will be done by functionnal filtering content, @itkg-alavieille

File: Repository/AbstractAggregateRepository.php [hunk 4]
```diff
@@ -25,6 +25,30 @@ public function setAggregationQueryBuilder($aggregationQueryBuilder)
...
+
+        return $return['retval'];
```
Comment (alavieille) [inline Repository/AbstractAggregateRepository.php]: may be test if index retval exist and return null if not exist

**Comment by alavieille** [comment]
:+1:

File: Repository/AbstractAggregateRepository.php [hunk 5]
```diff
@@ -25,6 +25,34 @@ public function setAggregationQueryBuilder($aggregationQueryBuilder)
...
+     */
+    public function findWithTransverseCriteria($criteria)
+    {
+        return $this->findBy($this->createTransverseCriteria($criteria));
+    }
+
+    /**
+     * @param string $criteria
+     *
+     * @return string
```
Comment (nicolasThal) [inline Repository/AbstractAggregateRepository.php]: are you sure that it is only string the return?

**Comment by alavieille** [comment]
@nicolasThal @itkg-rboessel

**Comment by nicolasThal** [comment]
:+1:

**Pull request merged**